### DISCUSSION TITLE
GRID_STAT_CLIMO_MEAN_FIELD WARNING in EMC EVS Testing

### DISCUSSION THREAD

**Discussion by ShannonShields-NOAA** [dtcenter/METplus#2554] [Use Cases] [answered]
Hello, this might have been missed in our original EMC EVS testing of the METv12.0.0-beta3 and METplusv6.0.0-beta3, so I want to make sure we are all aware of a possible bug I found in the code. I know we brought up "WARNING: Both *_STAT_CLIMO_MEAN_INPUT_TEMPLATE and *_STAT_CLIMO_MEAN_FILE_NAME are set. Using value set in *_STAT_CLIMO_MEAN_FILE_NAME." However, in the EMC EVS parallel output /lfs/h2/emc/ptmp/emc.vpppg/output/jevs_subseasonal_gefs_grid2grid_stats.o142676454 I am also seeing:
04/17 17:03:27.681 metplus.9e2bc524 (met_config.py:959) WARNING: Both GRID_STAT_CLIMO_MEAN_**FIELD** and GRID_STAT_CLIMO_MEAN_**VAR<n>**_* variables are set. Using values set in GRID_STAT_CLIMO_MEAN_VAR<n>_* variables
This seems like a similar bug as the other warning message because I am not setting GRID_STAT_CLIMO_MEAN_FIELD in my config file and yet, the warning says I am. Oddly enough, it seems the warning is fixed if I have GRID_STAT_CLIMO_MEAN_FIELD =
I was wondering if DTC is aware of this potential bug? Separate from the other CLIMO_MEAN bug that was found earlier?

**Comment by CPKalb** [comment]
Hi Shannon:

There are a number of variables in METplus that behave as you mentioned above.  Specifically, they have to be set with nothing after the = sign otherwise the value gets set to something unexpected.  

@georgemccabe, @JohnHalleyGotway, @fisherhucar can any of you verify if GRID_STAT_CLIMO_MEAN_FIELD has to be set to empty as described above (GRID_STAT_CLIMO_MEAN_FIELD = ) otherwise it gets set to something?  Is this the way this variable behaves?

Christina

**Comment by georgemccabe** [comment, marked as answer]
Hi @ShannonShields-NOAA, This is related to the same bug that involves the INPUT_TEMPLATE and FILE_NAME variables. The wrappers were being initialized twice and the 2nd pass was incorrectly outputting the warning because the 1st pass sets `GRID_STAT_CLIMO_MEAN_FIELD`. This was fixed with PR #2522 and is available in the METplus 6.0.0-beta4 release. You should no longer see the warning message using that version or above.

**Comment by ShannonShields-NOAA** [comment]
Thank you George for confirming my thoughts on this bug. When beta4 is
installed on WCOSS2, I will be sure to test the fix.
Shannon

On Thu, Apr 18, 2024 at 5:47 PM George McCabe ***@***.***>
wrote:

> Hi @ShannonShields-NOAA <https://github.com/ShannonShields-NOAA>, This is
> related to the same bug that involves the INPUT_TEMPLATE and FILE_NAME
> variables. The wrappers were being initialized twice and the 2nd pass was
> incorrectly outputting the warning because the 1st pass sets
> GRID_STAT_CLIMO_MEAN_FIELD. This was fixed with PR #2522
> <https://github.com/dtcenter/METplus/pull/2522> and is available in the
> METplus 6.0.0-beta4 release. You should no longer see the warning message
> using that version or above.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/discussions/2554#discussioncomment-9159919>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP4MEUA54AJ3SCIF4XKGV6DY6A5QLAVCNFSM6AAAAABGL7UVWGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNJZHEYTS>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Shannon Shields
SAIC at NOAA/NWS/NCEP/EMC
5830 University Research Ct. Rm 2180
College Park, MD 20740
***@***.***
